schema_based_on_diagram.sql

```diff
@@ -0,0 +1,44 @@
+-- Table: Patient
+CREATE TABLE Patient (
+  id INT PRIMARY KEY,
+  name VARCHAR(100),
+  date_of_birth DATE
+);
+
+-- Table: Medical_Histories
+CREATE TABLE Medical_Histories (
+  id INT PRIMARY KEY,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR(100),
+  FOREIGN KEY (patient_id) REFERENCES Patient(id)
+);
+
+-- Table: Invoices
+CREATE TABLE Invoices (
+  id INT PRIMARY KEY,
+  total_amount DECIMAL(10, 2),
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT,
+  FOREIGN KEY (medical_history_id) REFERENCES Medical_Histories(id)
+);
+
+-- Table: Treatments
+CREATE TABLE Treatments (
+  id INT PRIMARY KEY,
+  type VARCHAR(100),
+  name VARCHAR(100)
+);
+
```

Comment:
Hi,

You have done a good job with creating the tables and setting the primary and foreign keys. However, there are some issues and improvements that you can make to your code. Here are some suggestions:

- The `medical_histories_has_treatments` table is missing in your code. This table is needed to create a many-to-many relationship between `medical_histories` and `treatments`. Without this table, you will not be able to store the information about which treatments are applied to which medical histories.

Reply:
Hi @topeogunleye,
Thank you for your feedback. I apologize for the confusion. Upon review, I realized that the omission of the `Patient_Treatments` and `Medical_History_Treatments` tables was due to my failure to push the changes to the repository. The necessary changes were made locally, and the missing join tables have now been included in this PR.

Reply:
@MussieTeka, Thank you for explaining. Please kindly submit for review once you're done.